pallets/loans/src/rate.rs

```diff
@@ -41,13 +42,22 @@ impl From<Rate> for APR {
 impl APR {
     pub const MAX: Rate = Rate::from_inner(350_000_000_000_000_000); // 35%
 
+    pub fn from_fraction(delta_time: Timestamp) -> Option<Rate> {
+        FixedU128::from(1)
+            .checked_mul(&Rate::saturating_from_integer(delta_time))?
+            .checked_div(&Rate::saturating_from_integer(SECONDS_PER_YEAR))
+    }
+
     pub fn rate_per_block(&self, block_per_year: u128) -> Option<Rate> {
         self.0
             .checked_div(&Rate::saturating_from_integer(block_per_year))
     }
 
-    pub fn accrued_interest_per_block(&self, amount: u128, block_per_year: u128) -> Option<u128> {
-        self.0.checked_mul_int(amount)?.checked_div(block_per_year)
+    pub fn accrued_interest_per_block(&self, amount: u128, delta_time: Timestamp) -> Option<u128> {
+        let fraction: Rate = Self::from_fraction(delta_time)?;
+        self.0
+            .checked_mul_int(amount)?
+            .checked_div(fraction.into_inner())
```

Comment:
What does this mean? Could you talk more about the functionality of this method?

Reply:
Just corrected it. As discussed with you, it should be

```
   pub fn accrued_interest_per_block(&self, amount: u128, delta_time: Timestamp) -> Option<u128> {
        let fraction: Rate = Self::from_fraction(delta_time)?;
        self.0
            .checked_mul_int(amount)?
            .checked_mul(fraction.into_inner())
    }
```

Regarding `fraction.into_inner()`, I am converting FixedU128 into u128.